Controllers/ToDoController.cs

```diff
@@ -0,0 +1,286 @@
+
+            var todos = _context.ToDo.ToList();
+
+            if (!string.IsNullOrEmpty(searchString))
+            {
+                todos = todos.Where(t => t.Title.Contains(searchString)).ToList();
+            }
+
+            switch (sortOrder)
+            {
+                case "Title":
+                    todos = todos.OrderBy(t => t.Title).ToList();
+                    break;
+                case "title_desc":
+                    todos = todos.OrderByDescending(t => t.Title).ToList();
+                    break;
+                case "IsCompleted":
+                    todos = todos.OrderBy(t => t.IsCompleted).ToList();
+                    break;
+                case "completed_desc":
+                    todos = todos.OrderByDescending(t => t.IsCompleted).ToList();
+                    break;
+                case "CreatedDate":
+                    todos = todos.OrderBy(t => t.CreatedDate).ToList();
+                    break;
+                case "created_desc":
+                    todos = todos.OrderByDescending(t => t.CreatedDate).ToList();
+                    break;
+                case "UpdatedDate":
+                    todos = todos.OrderBy(t => t.UpdatedDate).ToList();
+                    break;
+                case "updated_desc":
+                    todos = todos.OrderByDescending(t => t.UpdatedDate).ToList();
+                    break;
+                default:
+                    todos = todos.OrderBy(t => t.Title).ToList();
+                    break;
+            }
+
+            var pageNumber = page ?? 1;
+            var pageSize = 5; 
+
+            IPagedList<ToDo> pagedToDos = todos.ToPagedList(pageNumber, pageSize);
+
+            return View(pagedToDos);
+        }
+
+        // GET: ToDo/Details/5
+        public async Task<IActionResult> Details(Guid? id)
+        {
+            if (id == null || _context.ToDo == null)
+            {
+                return NotFound();
+            }
+
+            var toDo = await _context.ToDo
+                .FirstOrDefaultAsync(m => m.Id == id);
+            if (toDo == null)
+            {
+                return NotFound();
+            }
+
+            return View(toDo);
+        }
+
+        // GET: ToDo/Create
+        public IActionResult Create()
+        {
+            return View();
+        }
+
+        // POST: ToDo/Create
+        // To protect from overposting attacks, enable the specific properties you want to bind to.
+        // For more details, see http://go.microsoft.com/fwlink/?LinkId=317598.
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public async Task<IActionResult> Create([Bind("Id,Title,IsCompleted,CreatedDate,UpdatedDate")] ToDo toDo)
+        {
+            if (ModelState.IsValid)
+            {
+                toDo.Id = Guid.NewGuid();
+                _context.Add(toDo);
+                await _context.SaveChangesAsync();
```

Comment:
if an error occurs what will you do on savechanges ?? please handle exception

Reply:
Done